src/main/kotlin/com/coxautodev/graphql/tools/SchemaParserBuilder.kt

```diff
@@ -280,7 +283,23 @@ data class SchemaParserOptions internal constructor(val genericWrappers: List<Ge
         }
     }
 
-    data class GenericWrapper(val type: Class<*>, val index: Int) {
-        constructor(type: KClass<*>, index: Int): this(type.java, index)
+    data class GenericWrapper(val type: Class<*>, val index: Int, val transformer: (Any) -> Any?) {
```

Comment:
Could we default `transformer` using [Kotlin's default parameters](https://kotlinlang.org/docs/reference/functions.html) rather than using overloaded constructors, Java-style?

Reply:
I just gave it a try, but the test _"parser should allow setting custom generic wrappers"_ is now failing:

```
Caused by: groovy.lang.GroovyRuntimeException: Could not find matching constructor for: com.coxautodev.graphql.tools.SchemaParserOptions$GenericWrapper(java.lang.Class, java.lang.Integer)
```

Any idea what's happening here?

Reply:
@sp00m I think you're getting that message because kotlin doesn't generate overloaded methods for java by default.  Since groovy compiles against the java class and not the kclass, it can't find the right constructor.  If you annotate the constructor with `@JvmOverloads` it should work.